### PR TITLE
[sample_consensus] Use better types for indices: `int` -> `index_t`, `std::vector<int>` ->`Indices`

### PR DESCRIPTION
**Pull request by mvieth**
Sorry for the big commit, I didn't see a way to divide that logically.

### CODE REVIEW

**Comment by mvieth** [comment]
I think `std::size_t` is used nowhere as a type for indices in sample consensus, is it?

**Comment by kunaltyagi** [comment]
> is it?

I don't know, sorry. Just something that popped in my mind. I'll have better access to PCL code in sometime when I'll be able to grok and tell you but you might be able to do that by the time I provide a yay/nay.

File: sample_consensus/include/pcl/sample_consensus/impl/prosac.hpp [hunk 1]
```diff
@@ -164,7 +164,7 @@ pcl::ProgressiveSampleConsensus<PointT>::computeModel (int debug_verbosity_level
 
       // We only need to compute possible better epsilon_n_star for when _n is just about to be removed an inlier
       std::size_t I_possible_n_star = I_N;
-      for (std::vector<int>::const_reverse_iterator last_inlier = inliers.rbegin (), 
+      for (Indices::const_reverse_iterator last_inlier = inliers.rbegin (),
```
Comment (taketwo) [inline sample_consensus/include/pcl/sample_consensus/impl/prosac.hpp:167]: I'd prefer `auto` for iterators.

**Comment by taketwo** [comment]
@kunaltyagi Sorry I'm gonna bug you with some changelog-related sidenotes again :grimacing: Why do you add `[module]` prefix to the title? We already have a label for that!

**Comment by kunaltyagi** [comment]
> Why do you add [module] prefix to the title?

* `[XYZ]` is shorter and less ambiguous than alternatives like `in XYZ` or `in module XYZ`
* more readable while reviewing PR/looking at the notification

I don't add it for the changelog..

**Comment by taketwo** [comment]
OK, I see the point. Still, as it stands, the line in changelog will have module prefix twice (once from the PR title, and once generated from labels). I see two options: a) we modify changelog generation script to strip module prefixes from titles; b) we modify title after a PR is merged and is dormant (not generating new notifications in your inbox). I guess the former is better.

> I don't add it for the changelog...

Didn't get what you mean with this.

**Comment by kunaltyagi** [comment]
> Didn't get what you mean with this.

I add the [module] for the notifications (same as issues). Didn't think of the changelog format.

Stripping the first `[]` sounds more doable to me than using a bot to do it after a PR is merged.

**Comment by taketwo** [comment]
> Stripping the first [] sounds more doable to me than using a bot to do it after a PR is merged.

Oh, I did not think about a bot, rather manual title change. But yeah, let's auto-strip on changelog generation. I will add this to my todo.

File: sample_consensus/include/pcl/sample_consensus/impl/prosac.hpp (hunk 1 above)
Comment (mvieth) [reply sample_consensus/include/pcl/sample_consensus/impl/prosac.hpp]: done

**Pull request merged**